compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala

```diff
@@ -19,6 +19,20 @@ class ErrorMessagesTests extends ErrorMessagesTest {
     checkMessagesAfter(FrontEnd.name)("""class Foo""")
     .expectNoErrors
 
+  @Test def caseClassExtendsEnum =
+    checkMessagesAfter(RefChecks.name) {
+      """
+        |enum Foo {}
+        |case class Bar() extends Foo
+      """.stripMargin
+    }
+      .expect { (ictx, messages) ⇒
+        implicit val ctx: Context = ictx
+        assertMessageCount(1, messages)
+        val errorMsg = messages.head
+        assertEquals("normal case class Bar in package <empty> cannot extend an enum", errorMsg.msg)
```

Comment:
FWIW: we usually don't test strings but that's the correct thing *as long as we don't switch to error messages here (#1589). @skvithalani are you interested in doing that in a followup? Accepting this meanwhile.

Reply:
Sure . I will take that up.

Reply:
Thanks! Beyond the pre-labeled issues, ones in parser/desugar tend to be more approachable.